pkg/virt-handler/vm.go

```diff
@@ -2047,6 +2053,10 @@ func (d *VirtualMachineController) processVmUpdate(origVMI *v1.VirtualMachineIns
 			if err != nil {
 				return fmt.Errorf("failed to adjust resources: %v", err)
 			}
+		} else if vmi.IsRunning() && !vmi.IsFinal() {
+			if err := d.hotplugVolumeMounter.Mount(vmi); err != nil {
+				return err
+			}
```

Comment:
I think it would be worth placing both the dynamic mount and unmount logic here.

This would mean anything that is involved with the hotplug of a vmi at runtime would exist here within this ```if vmi.Running()``` statement, and the "umount everything after shutdown" logic would be in the  ```processVmCleanup()```

Reply:
Okay sounds good.